### PR TITLE
Curl use CURL_INOUT events

### PR DESCRIPTION
**Pull request by jbaldwin**

### CODE REVIEW

File: examples/async_post_requests.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    while (count < 100)
+    {
+        ++count;
+
+        auto request = request_pool.Produce(host, on_complete, 60'000ms);
+        request->SetRequestData(data);
+        request->SetMethod(lift::http::Method::POST);
+        request->SetFollowRedirects(true);
+        request->SetVersion(lift::http::Version::V1_1);
+//        request->AddHeader("Expect", "");
```
Comment (PrestonFrom) [inline examples/async_post_requests.cpp]: I think I forgot to do this, but do we want to uncomment this or remove it?

File: examples/async_post_requests.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    lift::EventLoop event_loop;
+    auto& request_pool = event_loop.GetRequestPool();
+
+    auto count = 0;
+
+    while (count < 100)
+    {
+        ++count;
+
+        auto request = request_pool.Produce(host, on_complete, 60'000ms);
```
Comment (PrestonFrom) [inline examples/async_post_requests.cpp]: Do we want to use a shorter timeout time or does it not matter because this is just an example?

Comment (jbaldwin) [reply examples/async_post_requests.cpp]: I think for the sake of the example its better to give a longer timeout, but the best would be to have it be a cmd line arg.

File: examples/async_post_requests.cpp (hunk 1 above)
Comment (jbaldwin) [reply examples/async_post_requests.cpp]: Either/or.  I was commenting / uncommenting this line to trigger the different behavior, can probably just leave it here for now

**Pull request merged**